File: backend/src/main/scala/microservices/attraction/objects/GetAttractionDetailsPlannerRequest.scala
package com.typesafe.travel.attraction.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class GetAttractionDetailsPlannerRequest(attractionId: String, useDate: Option[String])
object GetAttractionDetailsPlannerRequest:
  given sourceEncoder: Encoder[GetAttractionDetailsPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[GetAttractionDetailsPlannerRequest] = deriveDecoder

File: frontend/src/microservices/attraction/objects/GetAttractionDetailsPlannerRequest.ts
// 本文件定义 attraction 模块的 `GetAttractionDetailsPlannerRequest`，作为详情查询参数并提供 JSON 编解码。

export type GetAttractionDetailsPlannerRequest = {
  attractionId: string
  useDate?: string
}

export const getAttractionDetailsPlannerRequestFromJson = (json: string): GetAttractionDetailsPlannerRequest =>
  JSON.parse(json) as GetAttractionDetailsPlannerRequest

export const getAttractionDetailsPlannerRequestToJson = (value: GetAttractionDetailsPlannerRequest): string =>
  JSON.stringify(value)
